### ISSUE TITLE
Use bundler 1.16.1?

### ISSUE THREAD

**Issue by jaredbeck** [heroku/heroku-buildpack-ruby#655]
On 2017-12-04 [it was suggested to use bundler 1.16.0][1] but there were problems with that release.

> .. Bundler 1.16.0 is riddled with issues. For instance, it breaks inline Gemfile, which breaks the Rails test suite. We'll be waiting for 1.16.1 before we consider upgrading it in the buildpack.

On 2017-12-21 bundler 1.16.1 was released. I have no idea if it fixes the issues Terence is concerned about, but I thought I'd open an issue to remind us to review 1.16.1.

I'm not familiar with this repo, so I hope this is helpful. If not, please disregard.

[1]: https://github.com/heroku/heroku-buildpack-ruby/pull/638

**Comment by borja** [comment]
I agree. bundler v.1.16.1 may be used not only for ruby on rails... It would be nice to have it available for other ruby apps! Thanks!

**Comment by hone** [comment]
@borja are there specific things you need from 1.16.1?

my concerns rise around bundler's trampoline feature which is what broke rails in the 1st place.

**Comment by jaredbeck** [comment]
> are there specific things you need from 1.16.1?

Speaking for myself, no, nothing specific. In general, however, we all know that eventually bundler 1.15 will no longer receive support, so we must upgrade to 1.16 eventually, so we might as well talk about it.

**Comment by jaredbeck** [comment]
> my concerns rise around bundler's trampoline feature which is what broke rails in the 1st place.

AFAICT the trampoline feature was removed in June 2017, and released in 1.16.1.

> My diagnosis was our best bet was to completely remove the trampoline from bundler itself.
> .. completely delete all references to trampolining / postit from Bundler and admit defeat.
> -Samuel Giddins
> https://github.com/bundler/bundler/pull/5804

**Comment by mattheworiordan** [comment]
Unless I am mistaken, until this buildpack supports Bundler 1.16.*, we're mostly stuck with Ruby 2.4.*.  Can the maintainer please look at this?  I created #738, but it relies on a tarball in a location I cannot modify.

**Comment by bjeanes** [comment]
> are there specific things you need from 1.16.1?

A hard yes from me.

I have a _huge_ Rails codebase that I'm trying to upgrade to Rails 4. The size and scope of the endeavour precludes a long-lived branch so we are running the app with multiple Gemfiles and need to create a Rails 4 deployment of the WIP by specifying `BUNDLE_GEMFILE`. However, I'm being bitten by bundler/bundler#5815 (also mentioned in https://devcenter.heroku.com/articles/bundler-version#known-upgrade-issues).

Has 1.16.1 been tested already and shown to be incompatible with Rails as alluded earlier or is this still speculative?

**Comment by jaredbeck** [comment]
Closing due to inactivity.

**Comment by mattheworiordan** [comment]
@jaredbeck closed and not resolved in any way?

**Comment by jaredbeck** [comment]
> @jaredbeck closed and not resolved in any way?

Correct. Not resolved AFAIK. If someone else wants to open the issue, go for it. But, maybe this is something we'd get more action on by contacting customer service? idk

**Comment by mattheworiordan** [comment]
>  But, maybe this is something we'd get more action on by contacting customer service?

Well I would have thought they'd look at the issues.  But hey ho.  I suppose at some point someone else will hit this issue, and create a duplicate.  In the mean time, we're locked into older versions or Ruby sadly.

**Comment by bjeanes** [comment]
@jaredbeck I used to work at Heroku. IMO This issue should definitely be kept open, as it is an outstanding issue and people (myself included) are subscribed to it for updates. Duplicate issues helps no one and having it closed might falsely indicate to others that this is classified as wontfix which AFAIK, has not been the decision.

**Comment by jaredbeck** [comment]
>  IMO This issue should definitely be kept open, as it is an outstanding issue and people (myself included) are subscribed to it for updates.

OK, that makes sense. I was trying to clear out my list of issues going nowhere, but I guess I'll just re-open and unsubscribe instead.

I'd encourage anyone who cares about this issue to contact heroku customer service. I think we'll have more luck there.

**Comment by schneems** [comment]
Buildpack maintainer here. I've built bundler 1.16.2 and put it on our S3 bucket. You can use a fork of the buildpack if it's needed. In the mean time It would actually be helpful for me if a new fresh issue was created. This one came in while I was out. As I understand it 1.16.2 isn't generally fit for deployment across the platform, though I've not personally investigated this.

In general bundler is our most unstable dependency, its the only dependency for Ruby that has caused non-trivial downtime for a large number of customers. It's unfortunate that bugfixes are also shipped with changing behavior so we can't just rev the version at will.

**Comment by schneems** [comment]
To clarify why I said a new issue would be helpful. I'm still unclear after reading this thread exactly what issues people are hitting on the platform today that are fixed in 1.16.2. It's fine if we keep the conversation on this thread however there's a lot of backlog to read through and the use cases of upgrading to 1.16.2 are still unclear.

> Well I would have thought they'd look at the issues. 

@hone02 maintained this buildpack for 8 years was commenting on this thread while I was out.

>  until this buildpack supports Bundler 1.16., we're mostly stuck with Ruby 2.4.

Why?

> I have a huge Rails codebase that I'm trying to upgrade to Rails 4. The size and scope of the endeavour precludes a long-lived branch so we are running the app with multiple Gemfiles and need to create a Rails 4 deployment of the WIP by specifying BUNDLE_GEMFILE.

@bjeanes Hi 👋

In general `BUNDLE_GEMFILE` is not known to work on Heroku. I had a PR https://github.com/heroku/heroku-buildpack-ruby/pull/351 though it actually breaks a bunch of things. How are you currently deploying with different Gemfiles?

Maybe you can elaborate more on why 1.16.2 helps out with bundler/bundler#5815

> I'd encourage anyone who cares about this issue to contact heroku customer service. I think we'll have more luck there.

Not a bad idea. The ticket will eventually get to me and then I can look at the actual issues that your app is experiencing if you give me app access. If it's a "please do this ticket" with no actionable behavior in an application that needs to be addressed, then opening a ticket is not helpful.
